src/coordination/include/nuraft/coordinator_cluster_state.hpp

```diff
@@ -54,7 +61,8 @@ struct CoordinatorInstanceState {
 void to_json(nlohmann::json &j, ReplicationInstanceState const &instance_state);
 void from_json(nlohmann::json const &j, ReplicationInstanceState &instance_state);
 
-using TRaftLog = std::variant<CoordinatorToReplicaConfig, CoordinatorToCoordinatorConfig, std::string, utils::UUID>;
+using TRaftLog = std::variant<CoordinatorToReplicaConfig, std::string, utils::UUID, CoordinatorToCoordinatorConfig,
```

Comment:
To me looking at this type system, it smells that we need different types for std::string, utils::UUID and InstanceUpdate...

Reply:
If we can merge states, better.+ Not sure if variants are preferred for such larger types

Reply:
I am not sure, I did go with variant as it is basically union type, but I don't like this part either. I think we can fix it somehow, I need to see how

Reply:
I will open this in new PR to try to fix it, we can see what to do there